packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -466,7 +466,16 @@ const helperBehavior = {
 
         // The rest of this method will not happen until after a quote is received.
         // This may not happen until much later, when the market changes.
-        // eslint-disable-next-line no-await-in-loop
+        //
+        // This nested await is safe because "terminal-control-flow"
+        //
+        // It occurs at top level of the body of a top level terminal
+        // while. But do note that the enclosing function is an asynchronous
+        // generator function and that there is a `yield` within the loop.
+        // I don't think that causes any additional worries, but I have not
```

Comment:
I believe it does not. This must be used within a `for await`, which awaits at the top of the loop (and therefore at each `yield`). So a `yield` i effectively an `await`.